src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -598,6 +621,11 @@ public StructImpl download(
             defaultValue = "{}",
             named = true,
             doc = "An optional dict specifying authentication information for some of the URLs."),
+        @Param(
```

Comment:
Can we specify the generic types here?

Reply:
I am not sure if that would provide any value here. I just followed `authUnchecked` param which also doesn't specify these.